### PR TITLE
Test entire request

### LINKED ISSUE #252: Rest Input/Result test are not just about body
The automated test generate stubs for Body, but in RestJson and RestXml, Uri and Headers are important too. 

see for instance https://github.com/async-aws/aws/blob/master/src/Service/Lambda/tests/Unit/Input/AddLayerVersionPermissionRequestTest.php

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #252

Better review whith whitespace hidden

### CODE REVIEW

File: src/Service/S3/src/Input/PutObjectRequest.php [hunk 1]
```diff
@@ -519,7 +519,7 @@ public function getWebsiteRedirectLocation(): ?string
     public function request(): Request
     {
         // Prepare headers
-        $headers = ['content-type' => 'application/xml'];
+        $headers = [];
```
Comment (jderusse) [inline src/Service/S3/src/Input/PutObjectRequest.php]: That's a bug in the current implementation

File: src/Service/S3/tests/Unit/Input/PutObjectRequestTest.php [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 class PutObjectRequestTest extends TestCase
 {
-    public function testRequestBody()
+    public function testRequest()
     {
         $bucket = 'foo';
         $key = 'bar/baz/biz';
```
Comment (Nyholm) [inline src/Service/S3/tests/Unit/Input/PutObjectRequestTest.php]: These variables are only used once. Are they auto generated?

Comment (jderusse) [reply src/Service/S3/tests/Unit/Input/PutObjectRequestTest.php]: nop, did you try `git blame`?..

Comment (Nyholm) [reply src/Service/S3/tests/Unit/Input/PutObjectRequestTest.php]: They were used twice but this PR removed once instance. 

Im fine with this, I just wanted to make sure we dont generate variables that are only used once.

Comment (jderusse) [reply src/Service/S3/tests/Unit/Input/PutObjectRequestTest.php]: ohh. my bad. Removed

**Pull request merged**